### ISSUE TITLE
bug: ion-label not floating inside ion-item with select input on ion-modal dialog

### ISSUE THREAD

**Issue by olaisen81** [ionic-team/ionic-framework#23809]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

ion-label not floating inside ion-item with select input on ion-modal dialog, label overlay the select field value. 
Selecting a new value fix the overlay bug, however, it comes back reopening the dialog. 
The bug not exists on modeless views.

![2021-08-24_16-22-33](https://user-images.githubusercontent.com/20971348/130635982-5596650b-1709-499f-95a5-066a3b97e2da.png)

### Expected Behavior

Floating ion-label not overlay on select fields in both normal and ion.modal view. 

### Steps to Reproduce

Create a modal and open a modal with an ion-label with floating prop with an ion-select.

`<ion-item>
      <ion-label color="primary" position="floating">Posizione
      </ion-label>
      <ion-select multiple="false" formControlName="nascSesso"
        [(ngModel)]="this.entita.Collaboratore.Anagrafica.NascSesso"
        [cancelText]="this.translateService.instant('component.select.cancelText')" ok-text="Ok">
        <ion-select-option *ngFor="let item of this.tipoSesso | keys" [value]="item.key">
          {{ 'Youpass.model.anagrafica.EnumSesso.' + item.value | translate }}</ion-select-option>
      </ion-select>
    </ion-item>`

### Code Reproduction URL

https://github.com/olaisen81/ion-label-issue

### Ionic Info

![2021-08-24_16-24-22](https://user-images.githubusercontent.com/20971348/130635776-e9e07ee0-4cdf-4709-90cb-f1d814d47c23.png)

**Comment by olaisen81** [comment]
Ok, i've reproduced the bug upgrading angular to the latest version (12.2.2). With angular@12.1.5 the bug is not reproducing.
You can check the issue in my Github repo at https://github.com/olaisen81/ion-label-issue

**Comment by olaisen81** [comment]
Hi Liam! The same repository ( https://github.com/olaisen81/ion-label-issue) now have 2 branches, one named angular@latest and one angular@12.1.5, you can test now working and bugging version. Thanks for your reply!

**Comment by liamdebeasi** [comment]
Thanks! This behavior change looks to be caused by `@angular-devkit/build-angular@12.2.1` or newer. I have reported this issue to the Angular team here: https://github.com/angular/angular-cli/issues/21682

I will update this thread when I have more information. For now, you can work around this by staying on `@angular-devkit/build-angular@12.2.1`.

**Comment by olaisen81** [comment]
Great! Thanks Liam.

**Comment by liamdebeasi** [comment]
After some discussion with the Angular team, this looks like something we need to address inside of Ionic: https://github.com/angular/angular-cli/issues/21682

Ionic was leveraging a defect in zone.js which caused async operations to not be intercepted, resulting in this issue. It was technically an issue before this thread was opened, but we never noticed it until now because native async/await was not previously downlevelled.

I will look into a fix soon.

**Comment by realappie** [comment]
> After some discussion with the Angular team, this looks like something we need to address inside of Ionic: [angular/angular-cli#21682](https://github.com/angular/angular-cli/issues/21682)
> 
> Ionic was leveraging a defect in zone.js which caused async operations to not be intercepted, resulting in this issue. It was technically an issue before this thread was opened, but we never noticed it until now because native async/await was not previously downlevelled.
> 
> I will look into a fix soon.

Hey there, any upgrades on this fix  ? 👍

**Comment by olaisen81** [comment]
Unfortunately not yet.

**Comment by a2kan** [comment]
I applied a workaround to this problem for now, After setting the value of the select if you add "item-has-value" class to parent item element of the select in a setTimeout function, the things works somehow as expected.

@ViewChild('item', { static: false }) item: any;
..
 setTimeout(() => {
    this.item.el.classList.add('item-has-value');  
 }, 100);

 <ion-item #item>
  <ion-label position="floating">
              label
            </ion-label>
            <ion-select></ion-select>
</ion-item>

**Comment by sean-perkins** [comment]
Could someone that has encountered this issue in their application, please try with this dev build:
```bash
npm install @ionic/angular@6.0.0-dev.1641314109.170f765
```

and let me know if you notice any side effects? I've tested with the reproduction application and it appears to solve the visual issues.

Thanks!

**Comment by WolfWalter** [comment]
> Could someone that has encountered this issue in their application, please try with this dev build:
> 
> ```shell
> npm install @ionic/angular@6.0.0-dev.1641314109.170f765
> ```
> 
> and let me know if you notice any side effects? I've tested with the reproduction application and it appears to solve the visual issues.
> 
> Thanks!
Hi @sean-perkins,
I tired to use that version, but I got errors for every use of ngModel in my application:

```
[ng] Error: src/app/settings/settings.page.html:99:27 - error NG8007: The property and event halves of the two-way binding 'ngModel' are not bound to the same target.
[ng]             Find more at https://angular.io/guide/two-way-binding#how-two-way-binding-works
[ng] 
[ng] 99             <ion-toggle [(ngModel)]="changeInstallerRequestNotifications"></ion-toggle>
[ng]                              ~~~~~~~
[ng] 
[ng]   node_modules/@ionic/angular/directives/control-value-accessors/boolean-value-accessor.d.ts:4:22
[ng]     4 export declare class BooleanValueAccessorDirective extends ValueAccessor {
[ng]                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ng]     The property half of the binding is to the 'BooleanValueAccessorDirective' component.
[ng]   node_modules/@angular/forms/forms.d.ts:3617:22
[ng]     3617 export declare class NgModel extends NgControl implements OnChanges, OnDestroy {
[ng]                               ~~~~~~~
[ng]     The event half of the binding is to the 'NgModel' component.
[ng]   src/app/settings/settings.page.ts:25:16
[ng]     25   templateUrl: './settings.page.html',
[ng]                       ~~~~~~~~~~~~~~~~~~~~~~
[ng]     Error occurs in the template of component SettingsPage.
```

**Comment by sean-perkins** [comment]
Hello everyone, I have an updated dev build that solves the issue with less intense modifications.

Can someone validate with:
```
6.0.12-dev.11647441819.18de4788
```

This should work for both `ngModel` and reactive form usages.

**Comment by filipencus** [comment]
Hi @sean-perkins 
Do you know when this fix will be available for production? 
I checked all Ionic release history and I don't see anything about it.  Thanks.

**Comment by liamdebeasi** [comment]
This fix was released in 6.0.15: https://github.com/ionic-team/ionic-framework/blob/main/CHANGELOG.md#6015-2022-04-06

**Comment by filipencus** [comment]
Thanks for the info @liamdebeasi
